### PR TITLE
MEN-3676: Fix broken logging to syslogger.

### PR DESCRIPTION
**Pull request by kacf**
Changelog: Title

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

**Comment by lluiscampos** [comment]
> It seems like this has been broken for a long time.
> 
> Looks good, see comment below.

Well, the actual unit test for this is failing, so it "looks" good but something is wrong still.

File: cli/cli.go [hunk 1]
```diff
@@ -582,8 +582,7 @@ func (runOptions *runOptionsType) handleLogFlags(ctx *cli.Context) error {
 		}
 		log.SetOutput(fd)
 	}
-	if ctx.IsSet("no-syslog") &&
-		!runOptions.logOptions.noSyslog {
+	if !runOptions.logOptions.noSyslog {
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply cli/cli.go:585]: Okey, I see that (thanks), but then your fix could have been:

```
if !ctx.IsSet("no-syslog") && !runOptions.logOptions.noSyslog {
...
```

I don't think it is important: we are setting these logOptions from the args parsing for a single time use, so either is fine.

**Comment by kacf** [comment]
> Well, the actual unit test for this is failing, so it "looks" good but something is wrong still.

I don't know why this passes locally, but in any case the test is bogus. It tests whether the message reached *any* hook (in this case the test hook), but specifying -no-syslog only affects the syslog hook specifically. I just removed the the test, since the test in https://github.com/mendersoftware/meta-mender/pull/1017 covers it much better.

**Comment by Risca** [comment]
Yay! :smile:

**Pull request merged**